daphne_worker/src/dap.rs

```diff
@@ -251,11 +269,78 @@ where
         &self.global_config
     }
 
-    async fn get_task_config_for(
+    async fn get_task_config_considering_taskprov(
         &'srv self,
         task_id: Cow<'req, Id>,
+        metadata: Option<&ReportMetadata>,
     ) -> std::result::Result<Option<GuardedDapTaskConfig<'req>>, DapError> {
-        self.get_task_config(task_id).await.map_err(dap_err)
```

Comment:
I was a bit confused by the control flow because it seems more natural to check KV for an existing task config only if taskprov is disallowed. In fact, what we're doing here is detecting if we have already configured a taskprov task. Maybe add a comment to clarify?